bayesian/utils.py

```diff
@@ -455,3 +455,79 @@ def get_files_github_url(self, github_url):
     def del_temp_files(self):
         if os.path.exists(self.CLONED_DIR):
             shutil.rmtree(self.CLONED_DIR)
+
+
+class RecommendationReason:
+    """
+    This is used to provide the reason for alternate, and companion package recommendations.
+    """
+
+    def add_reco_reason(self, manifest_response):
+        """
+        It will populate english sentence for the recommendations.
+        :param manifest_response: dict. object having all recommendation elements
+        :return: same dict. object with populated reasons
+        """
+        # Populate reason for each companion package
+        manifest_response = self._companion_reason(manifest_response)
+
+        # Populate reason for each alternate package
+        manifest_response = self._alternate_reason(manifest_response)
+        return manifest_response
+
+    def _alternate_reason(self, manifest_response):
+        """
+        It will populate simple reason for each alternate package recommendation
+        :param manifest_response: dict. object having all recommendation elements
+        :return: same dict. object with populated reasons for alternate package
+        """
+        length = len(manifest_response[0].get("recommendation").get("alternate"))
+        if length > 0:
+            count = 0
+            while count < length:
```

Comment:
use a more pythonic way as
```
for pkg in manifest_response[0].get("recommendation", {}).get("alternate", []):
    name = pkg.get("name")
    replaces = pkg.get("replaces")
    test_usage_outlier = self._check_usage_outlier(name, manifest_response)
```
Change `test_usage_outlier` functions to get name instead of count.

Reply:
Done